openhands/runtime/plugins/agent_skills/utils/aider/linter.py

```diff
@@ -131,7 +139,100 @@ def check_ts_installed(self):
         except (subprocess.CalledProcessError, FileNotFoundError):
             return False
 
-    def ts_lint(self, fname, rel_fname, code):
+    def check_ts_installed(self) -> bool:
+        """Check if TypeScript is installed."""
+        return self._check_tool_installed('tsc')
+
+    def check_eslint_installed(self) -> bool:
+        """Check if ESLint is installed."""
+        return self._check_tool_installed('eslint')
+
+    def print_lint_result(self, lint_result):
+        if not isinstance(lint_result, LintResult):
+            return
+        print(f'\n{lint_result.text.strip()}')
+        if isinstance(lint_result.lines, list) and lint_result.lines:
+            if isinstance(lint_result.lines[0], LintResult):
+                self.print_lint_result(lint_result.lines[0])
+
+    def ts_eslint(self, fname, rel_fname, code):
+        """Use ESLint to check for errors. If ESLint is not installed return None."""
+        if not self.eslint_installed:
+            return None
+
+        # Enhanced ESLint configuration with React support
+        eslint_config = {
+            'env': {'es6': True, 'browser': True, 'node': True},
+            'extends': ['eslint:recommended', 'plugin:react/recommended'],
+            'parserOptions': {
+                'ecmaVersion': 2021,
+                'sourceType': 'module',
+                'ecmaFeatures': {'jsx': True},
+            },
+            'plugins': ['react'],
+            'rules': {
+                'no-unused-vars': 'warn',
+                'no-console': 'off',
+                'react/prop-types': 'warn',
+                'semi': ['error', 'always'],
+            },
+            'settings': {'react': {'version': 'detect'}},
+        }
+
+        # Write config to a temporary file
+        with tempfile.NamedTemporaryFile(
+            mode='w', suffix='.json', delete=False
+        ) as temp_config:
+            json.dump(eslint_config, temp_config)
+            temp_config_path = temp_config.name
```

Comment:
I'm a bit confused by this with syntax. Why not just use a normal with statement and put the remainder of the function inside the with statement? Then I think you wouldn't need all the `os.unlink`s